Usr/jasteph/win ui gs (#797)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/index.md

# Introduction to Microsoft Edge WebView2 (Preview)  

The Microsoft Edge WebView2 control enables you to embed web technologies \(HTML, CSS, and JavaScript\) in your native applications.  The WebView2 control uses [Microsoft Edge (Chromium)][MicrosoftedgeinsiderMain] as the rendering engine to display the web content in native applications.  With WebView2, you may embed web code in different parts of your native application, or build the entire native application within a single WebView.  For information on how to start building a WebView2 application, see [Get Started](#getting-started).  

:::image type="complex" source="./media/WebView2/whatwebview.png" alt-text="What is WebView" lightbox="./media/WebView2/whatwebview.png":::
   What is WebView  
:::image-end:::  

> [!NOTE]

> > The WebView2 Preview is intended for early prototyping and to gather feedback to help to shape the API.  The Microsoft Edge WebView team does not recommend that you use the preview in your production apps because there may be [breaking changes][Webview2Releasenotes].  

Reviewer: ```suggestion
> The WebView2 Preview is intended for early prototyping and to gather feedback to help shape the API.  We do not recommend that you use the preview in your production apps because there may be breaking changes. For more information, see [Webview2Releasenotes].  
```

## Getting started  

To build and test your application using the WebView2 control, you need to have both [Microsoft Edge (Chromium)][MicrosoftedgeinsiderDownload] and the [WebView2 SDK][NugetPackagesMicrosoftWebWebView2] installed.  Select one of the following options to get started.  

*   [Getting Started with Win32 C/C++][Webview2GettingstartedWin32]  
*   [Getting Started with WPF][Webview2GettingstartedWpf]  
*   [Getting Started with WinForms][Webview2GettingstartedWinforms]  
*   [Getting Started with WinUI3][Webview2GettingstartedWinui]  

> The [WebView2 Samples][GithubMicrosoftedgeWebview2samples] repository contains samples that demonstrate all of the WebView2 SDKs features and API usage patterns.  As more features are added to the WebView2 SDK, the sample applications is updated.  

Reviewer: ```suggestion
The [WebView2 Samples][GithubMicrosoftedgeWebview2samples] repository contains samples that demonstrate all of the WebView2 SDK features and API usage patterns.  As more features are added to the WebView2 SDK, the sample applications will be updated.  
```

*   Win32 C/C++  
*   .NET Framework 4.6.2 or later  
*   .NET Core 3.0 or later  

> *   [WinUI 3.0][UwpToolkitsWinui3]  

Reviewer: This link appears to be broken. Can you look into it please?

## Next steps  

> For more detailed information on how to build and deploy WebView2 applications, review the conceptual documentation and how-to guides.  

Reviewer: ```suggestion
For more information on how to build and deploy WebView2 applications, review the conceptual documentation and how-to guides.  
```

#### Concepts  

> *   [WebView2 SDK and Microsoft Edge Versioning][Webview2ConceptsVersioning]

Reviewer: ```suggestion
*   [Understand WebView2 SDK versions][Webview2ConceptsVersioning]
```

> *   [Distribute WebView2 Applications][Webview2ConceptsDistribution]  

Reviewer: ```suggestion
*   [Distribution of applications using WebView2][Webview2ConceptsDistribution]  
```

> *   [Security Best Practices for WebView2 Applications][Webview2ConceptsSecurity]

Reviewer: ```suggestion
*   [Best practices for developing secure WebView2 applications][Webview2ConceptsSecurity]
```

*   [Manage User Data Folder in WebView2 Applications][Webview2ConceptsUserdatafolder]
 
#### How-To guides  

> *   [Debugging WebView2 with DevTools and Visual Studio script debugging][Webview2HowtoDebug]  

Reviewer: ```suggestion
*   [How to Debug with WebView2][Webview2HowtoDebug]  
```

> *   [Automating and debugging WebView2 with Microsoft EdgeDriver][Webview2HowtoWebdriver]  

Reviewer: ```suggestion
*   [Automating and testing WebView2 with Microsoft Edge Driver][Webview2HowtoWebdriver]  
```

## Getting in touch with the WebView2 team  

> Help build a richer WebView2 experience by sharing your feedback.  to submit feature requests or bug reports, see [WebView feedback repo][GithubMicrosoftedgeWebviewfeddback] .  It is also a good place to search for known issues.  

Reviewer: ```suggestion
Help build a richer WebView2 experience by sharing your feedback.  To submit feature requests or bug reports, see [WebView feedback repo][GithubMicrosoftedgeWebviewfeddback] .  It's also a good place to search for known issues.  
```

> [!NOTE]

> > During developer preview, the Microsoft Edge WebView team also collects data to help build a better WebView.  To turn off WebView data collection, go to `edge://settings/privacy` and turn off browser data collection.  

Reviewer: ```suggestion
> During the preview, we collect data to help build a better product.  To turn off WebView2 data collection, go to `edge://settings/privacy` and turn off browser data collection.  
```
